addon/components/sl-modal.js

```diff
@@ -0,0 +1,236 @@
+import Ember from 'ember';
+import layout from '../templates/components/sl-modal';
+
+/**
+ * @module
+ */
+export default Ember.Component.extend({
+    // -----------------------------------------------------
+    // Dependencies
+
+
+    // -----------------------------------------------------
+    // Attributes
+
+
+    /** @type {String[]} */
+    attributeBindings: [
+        'aria-describedby',
+        'aria-hidden',
+        'aria-labelledby',
+        'tabindex'
+    ],
+
+    /** @type {String[]} */
+    classNames: [
+        'modal fade'
+    ],
+
+    /** @type {String} */
+    tagName: 'div',
+
+    /** @type {Object} */
+    layout: layout,
+
+
+    // -----------------------------------------------------
+    // Actions
+
+
+    /** @type {Object} */
+    actions: {
+        /**
+         * Trigger hiding the model
+         *
+         * @function actions:hide
+         * @returns {undefined}
+         */
+        hide() {
+            this.hide();
+        },
+
+         /**
+         * Trigger showing the model
+         *
+         * @function actions:show
+         * @returns {undefined}
+         */
+        show() {
+            this.show();
+        }
+    },
+
+
+    // -----------------------------------------------------
+    // Events
+
+
+    // -----------------------------------------------------
+    // Properties
+
+
+    /**
+     * aria-describedby attribute
+     *
+     * @type {?String}
+     * */
+    'aria-describedby': null,
+
+    /**
+     *  aria-hidden attribute to inform assistive technologies to skip the
+     *  modal's DOM elements
+     *
+     * @type {String}
+     *                          */
+    'aria-hidden': 'true',
+
+    /**
+     * The aria-role value
+     *
+     * @type {String}
+     *                     */
+    ariaRole: 'dialog',
+
+    /*
+     * Whether to show Bootstrap's backdrop
+     *
+     * @type {Boolean|String}
```

Comment:
I'm assuming that Twitter Bootstrap will accept either types but since we are explicitly setting it to a Boolean and are not supporting the overriding of it via a supplied property lets just set this to the single type of Boolean.